### PR TITLE
Async lookups-cached-global by default

### LINKED ISSUE #3063: Unloadable lookups can tie up all query threads
In 0.9.1-rc1, if the coordinator post to /druid/listen/v1 involves a lookup that can't be loaded, that will cause the request to block until the lookup load times out. After 10s the coordinator will time out this request. Later on it will issue a new one, but if the original request's thread is still active (which it totally could be), the new request's thread will just block while trying to acquire the lock held by the first request thread. Fast-forward a few minutes, and all jetty threads are blocked on lookup-related stuff and no queries can be made.

Possible solutions:

- Make /druid/listen/v1 requests nonblocking, by having them insert into a queue rather than actually do the work. The work would be done by another thread. The request thread would have to fail when the queue is full rather than block on full.
- Or if the requests really need to be blocking, limit the number of concurrent /druid/listen/v1 requests (by failing any requests if there are more than X in flight). X = 1 seems reasonable to me.

### PR DESCRIPTION
**Pull request by drcrallen**
* Also better lookup docs

Between this and #3068 I think this

fixes #3063

### CODE REVIEW

File: docs/content/development/extensions-core/lookups-cached-global.md [hunk 1]
```diff
@@ -70,9 +70,11 @@ The parameters are as follows
 |Property|Description|Required|Default|
 |--------|-----------|--------|-------|
 |`extractionNamespace`|Specifies how to populate the local cache. See below|Yes|-|
-|`firstCacheTimeout`|How long to wait (in ms) for the first run of the cache to populate. 0 indicates to not wait|No|`60000` (1 minute)|
+|`firstCacheTimeout`|How long to wait (in ms) for the first run of the cache to populate. 0 indicates to not wait|No|`0` (do not wait)|
 |`injective`|If the underlying map is injective (keys and values are unique) then optimizations can occur internally by setting this to `true`|No|`false`|
 
+If `firstCacheTimeout` is set to a non-zero value, it should be less than `druid.manager.lookups.hostUpdateTimeout`. If `firstCacheTimeout` is NOT set, then management is essentially asynchronous and does not know if a lookup succeeded or failed in starting. In such a case logs from the lookup nodes should be monitored for repeated failures.
```
Comment (gianm) [inline docs/content/development/extensions-core/lookups-cached-global.md]: This says "log should be monitored"; will they emit alerts? If not then that'd be useful.

**Comment by gianm** [comment]
@drcrallen I think as-is, setting firstCacheTimeout to 0 will cause the manager to report failure in `scheduleAndWait`, leading to a bunch of error messages and also to `delete(id)` getting called. If it's meant to act async-y it should report success even if the task hasn't finished yet.

**Comment by drcrallen** [comment]
@gianm that was missing a test. Fixed that case

**Comment by gianm** [comment]
👍  looks good!

File: docs/content/querying/lookups.md [hunk 2]
```diff
@@ -294,6 +294,16 @@ To configure a Broker / Router / Historical / Peon to announce itself as part of
 |`druid.lookup.lookupTier`| The tier for **lookups** for this node. This is independent of other tiers.|`__default`|
 |`druid.lookup.lookupTierIsDatasource`|For some things like indexing service tasks, the datasource is passed in the runtime properties of a task. This option fetches the tierName from the same value as the datasource for the task. It is suggested to only use this as peon options for the indexing service, if at all. If true, `druid.lookup.lookupTier` MUST NOT be specified|`"false"`|
 
+To configure the behavior of the dynamic configuration manager, use the following properties on the coordinator:
+
+|Property|Description|Default|
```
Comment (b-slim) [inline docs/content/querying/lookups.md]: not sure where those variable are used ?

Comment (b-slim) [inline docs/content/querying/lookups.md]: NVM i see it.

**Comment by b-slim** [comment]
👍

**Pull request merged**